devicetypes/smartthings/qubino-flush-thermostat.src/qubino-flush-thermostat.groovy

```diff
@@ -0,0 +1,325 @@
+			tileAttribute("device.thermostatMode", key: "PRIMARY_CONTROL") {
+				attributeState("off", icon: "st.thermostat.heating-cooling-off")
+				attributeState("heat", icon: "st.thermostat.heat")
+				attributeState("cool", icon: "st.thermostat.emergency-heat")
+			}
+			tileAttribute("device.temperature", key: "SECONDARY_CONTROL") {
+				attributeState("temperature", label:'${currentValue}°', icon: "st.alarm.temperature.normal",
+						backgroundColors:[
+								// Celsius
+								[value: 0, color: "#153591"],
+								[value: 7, color: "#1e9cbb"],
+								[value: 15, color: "#90d2a7"],
+								[value: 23, color: "#44b621"],
+								[value: 28, color: "#f1d801"],
+								[value: 35, color: "#d04e00"],
+								[value: 37, color: "#bc2323"],
+								// Fahrenheit
+								[value: 40, color: "#153591"],
+								[value: 44, color: "#1e9cbb"],
+								[value: 59, color: "#90d2a7"],
+								[value: 74, color: "#44b621"],
+								[value: 84, color: "#f1d801"],
+								[value: 95, color: "#d04e00"],
+								[value: 96, color: "#bc2323"]
+						]
+				)
+			}
+			tileAttribute("device.heatingSetpoint", key: "HEATING_SETPOINT") {
+				attributeState("default", label: '${currentValue}', unit: "°", defaultState: true)
+			}
+		}
+		controlTile("thermostatMode", "device.thermostatMode", "enum", width: 2 , height: 2, supportedStates: "device.supportedThermostatModes") {
+			state("off", action: "setThermostatMode", label: 'Off', icon: "st.thermostat.heating-cooling-off")
+			state("heat", action: "setThermostatMode", label: 'Heat', icon: "st.thermostat.heat")
+			state("cool", action: "setThermostatMode", label: 'Cool', icon: "st.thermostat.cool")
+		}
+		controlTile("heatingSetpoint", "device.heatingSetpoint", "slider",
+				sliderType: "HEATING",
+				debouncePeriod: 750,
+				range: "device.setpointRange",
+				width: 2, height: 2) {
+			state "default", action:"setHeatingSetpoint", label:'${currentValue}', backgroundColor: "#E86D13"
+		}
+		controlTile("coolingSetpoint", "device.coolingSetpoint", "slider",
+				sliderType: "COOLING",
+				debouncePeriod: 750,
+				range: "device.setpointRange",
+				width: 2, height: 2) {
+			state "default", action:"setCoolingSetpoint", label:'${currentValue}', backgroundColor: "#55D4ED"
+		}
+		standardTile("refresh", "command.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "refresh", label: 'refresh', action: "refresh.refresh", icon: "st.secondary.refresh-icon"
+		}
+		valueTile("power", "device.power", width: 2, height: 2) {
+			state "default", label:'${currentValue} W'
+		}
+		valueTile("energy", "device.energy", width: 2, height: 2) {
+			state "default", label:'${currentValue} kWh'
+		}
+		standardTile("changeMode", "device.changeMode", width: 2 , height: 2, inactiveLabel: false, decoration: "flat") {
+			state("default", action: "changeMode", label: 'Push to switch mode', nextState: "unpair")
+			state("unpair", label: 'Unpair and pair device again')
+		}
```

Comment:
this should be a preference instead

Why do they need to re-pair? Does updating `supportedThermostatModes` not work? And does the config value persist across remove/re-pair?

Reply:
or at least a preference as well as a tile

Reply:
Quote from device manual: 
“This parameter determines how the device will operate if it will operate in the heating mode or in the cooling mode. (...) After parameter change, first exclude device (without setting parameters to defaultvalue) and then re include the device!”
Device can't support three modes at single setup, we need to re-configure it if we want it to work in 'cool' mode ('heat' is default)

Reply:
I'm wondering if they say that in their documentation because they don't expect controllers to be able to handle it. It's entirely possible that it would work fine for us.

Reply:
I've tested it, and it works only using steps defined by Qubino in their manual. After changing parameter value and polling for Supported Modes there's no change. 
Only after re-pairing we're able to use another mode.